tsdb/tsm1/array_cursor_iterator.go

```diff
@@ -32,11 +32,11 @@ type arrayCursorIterator struct {
 	}
 }
 
-func (q *arrayCursorIterator) Next(ctx context.Context, r *tsdb.CursorRequest) (tsdb.Cursor, error) {
+func (q *arrayCursorIterator) Next(ctx context.Context, r *tsdb.CursorRequest) tsdb.Cursor {
 	q.key = tsdb.AppendSeriesKey(q.key[:0], r.Name, r.Tags)
 	id := q.e.sfile.SeriesIDTypedBySeriesKey(q.key)
```

Comment:
I dug into this a bit. Reading `SeriesIDTyped` instead of `SeriesID` throughout TSI and series file code is simple enough, but would take time to implement. We basically remove the `SeriesID` type and fix all the bugs by passing around `SeriesIDTyped` instead.

Ran into a wall with roaring bitmaps, however. Anywhere that we do union/intersection operations, roaring bitmaps are used, and the library we use is 32-bit only.

Reply:
The C implementation of roaring bitmaps has 64 bit support. A CGO wrapper exists, but it doesn't have the 64 bit parts added yet.
https://github.com/RoaringBitmap/gocroaring

Reply:
Great catch as to why it strips them; forgot about the 32-bit limitation for our roaring bitmaps.

Reply:
Java's 64 bit support is conceptually simple. Where the 32 bit roaring bitmap is a mapping from high 16 bits to containers of low 16 bits, the 64 bit roaring bitmaps is a mapping from high 32 bits to low 32 bits as roaring bitmaps. There are details to solve, of course.

https://github.com/RoaringBitmap/RoaringBitmap/blob/master/RoaringBitmap/src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java#L28-L31

Reply:
I seem to recall that one of the main reasons it's this way is because there isn't a 64-bit roaring bitmap Go library. A couple of times I thought about writing it but the priorities never lined up...

Reply:
#17143